### PR TITLE
File attachment/CRUD adapters for primary, derivatives

### PR DESCRIPTION
**Pull request by seanupton**
These commits introduce adapters around a work context to attach files (primary, derivative) and perform CRUD operations for both primary, derivative files.  This is related to functionality described in #57.

Attachment is (for primary, and optionally/recommended for derivative) an assign-and-commit system, where paths to files for attachment or removal are assigned to queues, and flushed as an effect of calling a `.commit!` method.

### CODE REVIEW

File: lib/newspaper_works/data/work_derivatives.rb [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+          orig_pos = file.tell
+          file.seek(0)
+          File.open(path, 'w') { |dstfile| dstfile.write(file.read) }
+          file.seek(orig_pos)
+        end
+        # finally, reload @paths after mutation
+        load_paths
+      end
+
+      # attach a single derivative file to work
```
Comment (ebenenglish) [inline lib/newspaper_works/data/work_derivatives.rb]: Is this comment correct? Is this method attaching a single derivative? Possible mismatch with method name `delete`.

File: spec/lib/newspaper_works/data/work_derivatives_spec.rb [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+  let(:encoded_text) do
+    file = Tempfile.new('txt_encoded.txt', encoding: 'UTF-8')
+    file.write('Gorgonzola Dolce® — on sale for £12.50/kg')
+    file.close
+    file.path
+  end
+
+  describe "enumerates available derivatives like hash" do
+    it "includes expected derivative path for work" do
+      adapter = described_class.new(work)
```
Comment (ebenenglish) [inline spec/lib/newspaper_works/data/work_derivatives_spec.rb]: Is there a way to DRY this line:
```ruby
adapter = described_class.new(work)
```
(It's used in most of the specs, it seems? Granted there are some permutations here and there.)

Or does `adapter` get mutated by some of the actions happening in the various specs (e.g. `mk_jp2_derivative`)?

File: lib/newspaper_works/data/work_derivatives.rb (hunk 1 above)
Comment (seanupton) [reply lib/newspaper_works/data/work_derivatives.rb]: Definitely a copy/paste issue.  Can correct comment.

File: spec/lib/newspaper_works/data/work_derivatives_spec.rb (hunk 2 above)
Comment (seanupton) [reply spec/lib/newspaper_works/data/work_derivatives_spec.rb]: Probably a `let` statement for all cases where `work` is being adapted, and not use of `bare_work`.  That would DRY enough to justify doing it.

**Comment by seanupton** [comment]
Commits added today address items found in review.

**Pull request merged**